arraycontext/container/dataclass.py

```diff
@@ -104,33 +114,46 @@ def is_array_field(f: Field) -> bool:
                 _BaseGenericAlias,
                 _SpecialForm,
             )
-            if isinstance(f.type, GenericAlias | _BaseGenericAlias | _SpecialForm):
+            if isinstance(field_type, GenericAlias | _BaseGenericAlias | _SpecialForm):
                 # NOTE: anything except a Union is not allowed
                 raise TypeError(
                         f"Typing annotation not supported on field '{f.name}': "
-                        f"'{f.type!r}'")
+                        f"'{field_type!r}'")
 
-            if not isinstance(f.type, type):
+            if not isinstance(field_type, type):
                 raise TypeError(
                         f"Field '{f.name}' not an instance of 'type': "
-                        f"'{f.type!r}'")
-
-        return is_array_type(f.type)
-
-    from pytools import partition
-    array_fields, non_array_fields = partition(is_array_field, fields(cls))
+                        f"'{field_type!r}'")
+
+        return is_array_type(field_type)
+
+    array_fields: list[Field] = []
+    non_array_fields: list[Field] = []
+    from inspect import get_annotations
+    cls_ann: Mapping[str, type] | None = None
```

Comment:
```suggestion
    from inspect import get_annotations
    
    array_fields: list[Field] = []
    non_array_fields: list[Field] = []
    cls_ann: Mapping[str, type] | None = None
```
? It was confusing to see the import in the middle there :D

Reply:
Fixed!